.travis.yml

```diff
@@ -1,14 +1,89 @@
 language: python
-python: 3.5
-sudo: false
-env:
-- TOXENV=py27
-- TOXENV=py33
-- TOXENV=py34
-- TOXENV=py35
+matrix:
+  include:
+  - os: linux
+    python: 2.7
+  - os: linux
+    python: 3.3
+  - os: linux
+    python: 3.4
+  - os: linux
+    python: 3.5
+  - os: osx
+    language: generic
+    env: PYTHON='3.6.0'
+  - os: osx
+    language: generic
+    env: PYTHON='3.5.2'
+  - os: osx
+    language: generic
+    env: PYTHON='2.7.12'
+  - sudo: required
+    services:
+      - docker
+    env: DOCKER_IMAGE=quay.io/pypa/manylinux1_x86_64
+  - sudo: required
+    services:
+      - docker
+    env: DOCKER_IMAGE=quay.io/pypa/manylinux1_i686
+         PRE_CMD=linux32
+
+before_install:
+- |
+  if [[ $PYTHON && "$TRAVIS_OS_NAME" == "osx" ]]; then
+    brew update;
+    brew install openssl readline;
+    brew outdated pyenv || brew upgrade pyenv;
+    brew install pyenv-virtualenv;
+    pyenv install $PYTHON;
+    export PYENV_VERSION=$PYTHON;
+    export PATH="/Users/travis/.pyenv/shims:${PATH}";
+    pyenv-virtualenv venv;
+    source venv/bin/activate;
+    python --version;
+  fi;
+  echo [distutils]                                  > ~/.pypirc;
+  echo index-servers=pypi                          >> ~/.pypirc;
+  echo [pypi]                                      >> ~/.pypirc;
+  echo repository=https://pypi.python.org/pypi     >> ~/.pypirc;
+  echo username=kmike                              >> ~/.pypirc;
+  echo password=$PYPIPASSWORD                      >> ~/.pypirc;
+  touch .travis_tag;
+  if [[ $DOCKER_IMAGE ]]; then
+    echo $TRAVIS_TAG > .travis_tag;
+    cat ~/.pypirc > .pypirc;
+  fi;
+
 install:
-- pip install tox
-- pip install -U cython
+- |
+  if [[ $DOCKER_IMAGE ]]; then
+    docker pull $DOCKER_IMAGE &&
+    docker run --rm -v `pwd`:/io $DOCKER_IMAGE $PRE_CMD /io/.manylinux-install.sh;
+    exit;
+  else
+    pip install --upgrade pip;
+    pip install -r requirements-doc.txt
+    pip install pytest
+    pip install -U cython
+    pip install argparse
+    ./update_cpp.sh;
+    pip install -e .;
+  fi;
+
 script:
-- ./update_cpp.sh
-- tox
+- py.test tests --doctest-modules
```

Comment:
I liked tox tests because with tox you can be sure the same test running method is used locally and on Travis. "matrix" covers testing for different Python versions on Travis, but not locally.